source/darwin/dispatch_queue_event_loop.c

```diff
@@ -570,28 +631,30 @@ static void s_run_iteration(void *service_entry) {
  * aws_dispatch_loop->sycned_data
  */
 static void s_try_schedule_new_iteration(struct aws_dispatch_loop *dispatch_loop, uint64_t timestamp) {
-    if (dispatch_loop->synced_data.suspended || dispatch_loop->synced_data.is_executing) {
-        return;
-    }
-
-    if (!s_should_schedule_iteration(&dispatch_loop->synced_data.scheduled_iterations, timestamp)) {
+    if (dispatch_loop->synced_data.execution_state != AWS_DLES_RUNNING) {
         return;
     }
 
-    struct scheduled_iteration_entry *entry = s_scheduled_iteration_entry_new(dispatch_loop, timestamp);
-    aws_priority_queue_push_ref(
-        &dispatch_loop->synced_data.scheduled_iterations, (void *)&entry, &entry->priority_queue_node);
-
     /**
      * Apple dispatch queue uses automatic reference counting (ARC). If an iteration is scheduled to run in the future,
      * the dispatch queue will persist until it is executed. Scheduling a block far into the future will keep the
-     * dispatch queue alive unnecessarily long, even after aws_event_loop and aws_dispatch_loop have been fully
-     * destroyed and cleaned up. To mitigate this, we ensure an iteration is scheduled no longer than 1 second in the
+     * dispatch queue alive unnecessarily long, which blocks event loop group shutdown from completion.
+     * To mitigate this, we ensure an iteration is scheduled no longer than 1 second in the
```

Comment:
Trivial: Maybe change this comment to clamping to "AWS_DISPATCH_QUEUE_MAX_FUTURE_SERVICE_INTERVAL" since it's possible we change the time amount in the future?